src/Aspire.Hosting.Azure.ServiceBus/ServiceBusQueue.cs

```diff
@@ -0,0 +1,192 @@
+
+using System.Text.Json;
+using System.Xml;
+
+namespace Aspire.Hosting.Azure.ServiceBus;
+
+/// <summary>
+/// Represents a Service Bus Queue.
+/// </summary>
+/// <remarks>
+/// List of properties from Azure.Provisioning.ServiceBus that are not exposed here:
+/// - AutoDeleteOnIdle
+/// - EnableBatchedOperations
+/// - EnableExpress
+/// - EnablePartitioning
+/// - MaxMessageSizeInKilobytes
+/// - MaxSizeInMegabytes
+/// - Status
+///
+/// Use <see cref="AzureProvisioningResourceExtensions.ConfigureInfrastructure{T}(ApplicationModel.IResourceBuilder{T}, Action{AzureResourceInfrastructure})"/> to configure these specific properties.
+/// </remarks>
+public class ServiceBusQueue
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ServiceBusQueue"/> class.
+    /// </summary>
+    public ServiceBusQueue(string name)
+    {
+        Name = name;
+    }
+
+    /// <summary>
+    /// The queue name.
+    /// </summary>
+    public string Name { get; set; }
+
+    /// <summary>
+    /// A value that indicates whether this queue has dead letter support when
+    /// a message expires.
+    /// </summary>
+    public bool? DeadLetteringOnMessageExpiration { get; set; }
+
+    /// <summary>
+    /// ISO 8601 default message timespan to live value. This is the duration
+    /// after which the message expires, starting from when the message is
+    /// sent to Service Bus. This is the default value used when TimeToLive is
+    /// not set on a message itself.
+    /// </summary>
+    public TimeSpan? DefaultMessageTimeToLive { get; set; }
+
+    /// <summary>
+    /// ISO 8601 timeSpan structure that defines the duration of the duplicate
+    /// detection history. The default value is 10 minutes.
+    /// </summary>
+    public TimeSpan? DuplicateDetectionHistoryTimeWindow { get; set; }
+
+    /// <summary>
+    /// Queue/Topic name to forward the Dead Letter message.
+    /// </summary>
+    public string? ForwardDeadLetteredMessagesTo { get; set; }
+
+    /// <summary>
+    /// Queue/Topic name to forward the messages.
+    /// </summary>
+    public string? ForwardTo { get; set; }
+
+    /// <summary>
+    /// ISO 8601 timespan duration of a peek-lock; that is, the amount of time
+    /// that the message is locked for other receivers. The maximum value for
+    /// LockDuration is 5 minutes; the default value is 1 minute.
+    /// </summary>
+    public TimeSpan? LockDuration { get; set; }
+
+    /// <summary>
+    /// The maximum delivery count. A message is automatically dead-lettered
+    /// after this number of deliveries.
+    /// </summary>
+    public int? MaxDeliveryCount { get; set; }
+
+    /// <summary>
+    /// A value indicating if this queue requires duplicate detection.
+    /// </summary>
+    public bool? RequiresDuplicateDetection { get; set; }
+
+    /// <summary>
+    /// A value that indicates whether the queue supports the concept of
```

Comment:
Is it weird that this property is named "RequiresSession", but the docs say "whether the queue supports the concept of sessions"?

Reply:
This is coming from the CDK ... but independently of it being the correct description or not I wonder if instead of copying the value we could use the `<inheritdoc />`.